src/prototype.rs

```diff
@@ -138,34 +138,35 @@ impl<'a> FirstPass<'a> {
 
         self.finish_list(start_ix);
         let indent = line_start.scan_space_upto(4);
-        let remaining_space = line_start.remaining_space();
         if indent == 4 {
             let ix = start_ix + line_start.bytes_scanned();
+            let remaining_space = line_start.remaining_space();
             return self.parse_indented_code_block(ix, remaining_space);
         }
 
 
         // HTML Blocks
 
-        // Our HTML scanning functions don't know what to do with leading spaces,
-        // so start them out at the first nonspace character
-        let nonspace_ix = start_ix + line_start.bytes_scanned();
+        // Start scanning at the first nonspace character, but remember how many
+        // spaces there are because HTML blocks need to preserve them.
+        let remaining_space = indent;
```

Comment:
This will have, I think, the slight disadvantage of not including the leading spaces on the first line in the sourcemap. But it's such a tiny nit I wouldn't block on it.